api/v1alpha4/azuremachine_types.go

```diff
@@ -108,6 +108,10 @@ type AzureMachineSpec struct {
 	// SecurityProfile specifies the Security profile settings for a virtual machine.
 	// +optional
 	SecurityProfile *SecurityProfile `json:"securityProfile,omitempty"`
+
+	// SubnetName selects the Subnet where the VM will be placed
+	// +optional
+	SubnetName string `json:"subnetName,omitempty"`
```

Comment:
This field is optional so that the code keeps working with existing manifests. In the future we may think about changing it to be required?

Reply:
should we add validation to make it required when there is more than one subnet specified? otherwise it could be ambiguous

Reply:
Just to clarify: is your concern addressed by this? https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1411/files#diff-b6931ac61f37a13f66fc7836e039bf714a0553d5db4f313cc8ef31ca5f5647b9R218-R220